src/backend/columnar/cstore_writer.c

```diff
@@ -644,7 +647,7 @@ SerializeChunkData(TableWriteState *writeState, uint32 chunkIndex, uint32 rowCou
 		if (compressed)
 		{
 			serializedValueBuffer = compressionBuffer;
-			actualCompressionType = COMPRESSION_PG_LZ;
+			actualCompressionType = requestedCompressionType;
```

Comment:
No observable change here, this change and the change in the `Assert()` just makes code less pglz dependent, and adding LZ4 & zstd easier.